### PR TITLE
nickytonline/comment subscription fix

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Updated the server-side rendered (SSR) comment subscription button so that Preact VDOM diffing can replace the SSR button.

## QA Instructions, Screenshots, Recordings

Before

image

After

Jun-18-2020 11-22-04

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Johnny Rose: "The sinnk doesn't fix itself!"

### CODE REVIEW

File: app/javascript/packs/articlePage.jsx [hunk 1]
```diff
@@ -48,7 +48,6 @@ const userDataIntervalID = setInterval(async () => {
           onUnsubscribe={subscriptionRequestHandler}
         />,
         root,
-        root.firstElementChild,
```
Comment (nickytonline) [inline app/javascript/packs/articlePage.jsx:51]: This creeped in from a rebase I imagine. This is no longer required in Preact 10.x +

**Comment by nickytonline** [comment]
Odd thing is this isn't happening in production... Gonna test locally again before we merge this.

File: app/views/articles/_full_comment_area.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <% if @article.show_comments %>
       <header class="relative flex justify-between items-center">
         <h3 class="fs-2xl fw-bold">Discussion</h3>
-        <div id="comment-subscription"><span class="crayons-btn crayons-btn--outlined">Subscribe</span></div>
```
Comment (ludwiczakpawel) [inline app/views/articles/_full_comment_area.html.erb:6]: It was supposed to be a placeholder for this button until it's loaded so we don't have weird jump there. Is that neecessary to remove it?

Comment (nickytonline) [reply app/views/articles/_full_comment_area.html.erb:6]: Well on my local I see two buttons. But it doesn't appear to be happening in production. Just testing locally again...

Comment (nickytonline) [reply app/views/articles/_full_comment_area.html.erb:6]: Ahh, it looks like the Preact upgrade isn't in production yet. That's why.

Comment (nickytonline) [reply app/views/articles/_full_comment_area.html.erb:6]: We'll merge this and I'm investigating the fix to bring the SSR button back.

File: app/views/articles/_full_comment_area.html.erb [hunk 3]
```diff
@@ -3,7 +3,11 @@
     <% if @article.show_comments %>
       <header class="relative flex justify-between items-center">
         <h3 class="fs-2xl fw-bold">Discussion</h3>
-        <div id="comment-subscription"></div>
+        <div id="comment-subscription">
```
Comment (nickytonline) [inline app/views/articles/_full_comment_area.html.erb:6]: This allows Preact to VDOM diff correctly and replace the SSR button @ludwiczakpawel .

File: app/views/articles/_full_comment_area.html.erb (hunk 2 above)
Comment (nickytonline) [reply app/views/articles/_full_comment_area.html.erb:6]: All sorted out in this commit, https://github.com/thepracticaldev/dev.to/pull/8768/commits/e6ad10106c19310e09185eb2a0dfc6aa0d8e4b8c

**Pull request merged**